pkg/cmd/pulumi/stack/stack_export.go

```diff
@@ -106,21 +106,14 @@ func newStackExportCmd() *cobra.Command {
 					return stack.FormatDeploymentDeserializationError(err, stackName)
 				}
 
-				serializedDeployment, err := stack.SerializeDeployment(ctx, snap, true)
+				deployment, err = stack.SerializeUntypedDeployment(ctx, snap, &stack.SerializeOptions{
+					ShowSecrets: true,
+					Pretty:      true,
+				})
 				if err != nil {
 					return err
 				}
 
-				data, err := json.Marshal(serializedDeployment)
```

Comment:
This is `json.Marshal` so shouldn't pretty be false?

Reply:
Good spot, I meant to call this out as a deliberate change. This should fix a subtle bug we have here: if you pass `--show-secrets`, currently HTML characters are going to be escaped when we do this roundtrip, which isn't the behavior we want. This change should make it behave the way we intend: to not have those HTML characters escaped.

Note this only helps with the diy backend, as it marshals to JSON using `SetEscapeHTML(false)` for the result of `backend.ExportStackDeployment`. The httpstate backend uses `json.Marshal` from the service, so the HTML chars are always escaped there (even with the encoder used below on line 121).

Reply:
This is quite subtle and a good explanation here.  It's probably worth turning this into a in-code comment for future readers, as it's not obvious why we need `Pretty` set here.